### PR TITLE
Add support for value:toBalString()

### LINKED ISSUE #25765: Add new value:toBalString() method
**Description:**
$title according to https://github.com/ballerina-platform/ballerina-spec/issues/499

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by lasinicl**
Fixes #25765

```
xml argXML1 = xml `<TITLE>Empire Burlesque</TITLE>`;
argXML1.toBalString() \\xml`<TITLE>Empire Burlesque</TITLE>`
```
```
float?[] arr = [1.25, (0.0/0.0), (1.0/0.0), ()];
arr.toBalString() \\[1.25,float:NaN,float:Infinity,()]
```
```
decimal decimalVal = 1.20;
decimalVal.toBalString() \\1.20d
```
```
type Employee record {
    readonly int id;
    int age;
    decimal salary;
    readonly string name;
    boolean married;
};

table<Employee> employeeTable = table key(id, name) [
     { id: 1, age: 30,  salary: 300.5, name: "Mary", married: true },
     { id: 2, age: 20,  salary: 200, name: "John", married: true }
];
employeeTable.toBalString() \\table key(id,name) [{"id":1,"age":30,"salary":300.5d,"name":"Mary","married":true},{"id":2,"age":20,"salary":200d,"name":"John","married":true}]
```
```
public type FirstError distinct error;

error cause = error("Account not found", accountID = 123, name="someVal", oop = (), val=(0.0/0.0));
FirstError e = FirstError(REASON_1, cause, message = "Test union of errors with type test");
cause.toBalString() \\error error ("Account not found",accountID=123,name="someVal",oop=(),val=float:NaN)
e.toBalString() \\error FirstError ("Reason1",error error ("Account not found",accountID=123,name="someVal",oop=(),val=float:NaN),message="Test union of errors with type test")
```
object with toString method
```
public class Teacher {

    string name;
    string school;

    public function init(string name, string school) {
        self.name = name;
        self.school = school;
    }

    public function getDetails() returns string {
        return self.name + " from " + self.school;
    }

    public function toString() returns string {
        return self.getDetails();
    }
}

Teacher varObj = new("Rola", "MMV");
varObj.toBalString() \\object Rola from MMV
```
object without toString method
```
class Person {
    public int age = 0;
    public string name = "";
}

Person p1 = new;
p1.toBalString() \\ object Person1 986212188
```
```
Employee e1 =  { id: 1, age: 30,  salary: 300.5, name: "Mary", married: true };
e1.toBalString() \\{"id":1,"age":30,"salary":300.5d,"name":"Mary","married":true}
```

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/CycleUtils.java [hunk 1]
```diff
@@ -33,10 +33,12 @@
     public static class Node implements BLink {
         Object obj;
         BLink parent;
+        int index = 0;
 
         public Node(Object obj, BLink parent) {
             this.obj = obj;
             this.parent = parent;
+            this.index++;
```
Comment (gimantha) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/CycleUtils.java:41]: i think this should be this.index = this.parent.index + 1; ?

**Review by mohanvive — APPROVED** [review]

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/api/BStringUtils.java [hunk 2]
```diff
@@ -244,6 +245,83 @@ public static String getStringValue(Object value, BLink parent) {
...
+     * @return String value of the value in expression style
+     */
+    public static String getExpressionStringValue(Object value, BLink parent) {
+        if (value == null) {
+            return "()";
+        }
+
+        BType type = TypeChecker.getType(value);
+
+        //TODO: bstring - change to type tag check
```
Comment (gimantha) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/api/BStringUtils.java:262]: Lets do this change if not done yet

Comment (lasinicl) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/api/BStringUtils.java:262]: Added in https://github.com/ballerina-platform/module-ballerinai-transaction/pull/40/commits/ba3852ff7b6fab739a754330297a06682f4adab4

Comment (lasinicl) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/api/BStringUtils.java:262]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/27492/commits/a69f06a43a276d1e0ae5a8e61a94f9fe317f59c5

**Pull request merged**